src/textual/pilot.py

```diff
@@ -53,6 +53,7 @@ async def wait_for_animation(self) -> None:
     async def wait_for_scheduled_animations(self) -> None:
         """Wait for any current and scheduled animations to complete."""
         await self._app.animator.wait_until_complete()
+        await wait_for_idle(0)
```

Comment:
Unrelated to reactive stuff. This test was flaky. I think because even if the animator is done, there may still be pending messages.